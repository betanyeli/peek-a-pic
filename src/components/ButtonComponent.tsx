import React from 'react';
import { Button } from 'react-native-paper';

interface Props {
    onPress: () => void;
    icon: string;
}

export function ButtonComponent({
    onPress,
    children,
    icon = "camera"
}: React.PropsWithChildren<Props>) {
    return (
        <Button icon={icon} mode="contained-tonal" onPress={onPress} style={{ margin: 16 }}
        >
            {children}
        </Button>
    );
}


