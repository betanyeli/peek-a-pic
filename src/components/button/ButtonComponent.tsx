import React from 'react';
import { Button } from 'react-native-paper';

interface Props {
    onPress: () => void;
    icon: string;
    testID?: string;
}

export function ButtonComponent({
    onPress,
    children,
    icon = "camera",
    testID = "button"
}: React.PropsWithChildren<Props>) {
    return (
        <Button testID={testID} icon={icon} mode="contained-tonal" onPress={onPress} style={{ margin: 16 }}
        >
            {children}
        </Button>
    );
}


