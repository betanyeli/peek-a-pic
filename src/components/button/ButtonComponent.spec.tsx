import { ButtonComponent } from "./ButtonComponent";
import { fireEvent, render, waitFor } from '@testing-library/react-native';

describe('Button component test', () => {
    it('Should be match with snapshot', () => {
        //Arrange
        const fn = jest.fn()
        const { toJSON } = render(<ButtonComponent onPress={fn} icon={""} />)

        // Assert

        // Act
        waitFor(() => { expect(toJSON).toMatchSnapshot() });
    });

    it('Should trigger an OnPress Function when button is pressed', () => {
        //Arrange
        const fn = jest.fn()
        const { getByTestId } = render(<ButtonComponent onPress={fn} icon={""} />)
        const button = getByTestId('button');

        // Assert
        fireEvent.press(button)

        // Act
        expect(fn).toBeCalled()
    });
});