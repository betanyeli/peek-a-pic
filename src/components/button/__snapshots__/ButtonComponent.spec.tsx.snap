// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test Should be match with snapshot 1`] = `[Function]`;
